src/main/scala/higherkindness/rules_scala/workers/common/CommonArguments.scala

```diff
@@ -49,6 +49,19 @@ object Analysis {
 }
 
 object CommonArguments {
+  private def adjustCompilerOptions(workDir: Path, options: List[String]) = {
+    def adjustStringPath(path: String) = SandboxUtil.getSandboxPath(workDir, Paths.get(path)).toString
+
+    options.flatMap {
+      case s"-P:semanticdb:targetroot:$path" =>
```

Comment:
Are there other other common compiler options that we know of which need a specific path?

This is a tricky issue because in theory there are infinite options which could do so. Even better yet they get specified by the user.

I suppose it's probably just not work thinking about for now.

Reply:
> This is a tricky issue because in theory there are infinite options which could do so. Even better yet they get specified by the user.

I thought of this when I made this change and did feel slimy modifying compiler options like this, but I think it's better to modify compiler options than pass SemanticDB options separately. My reasoning is that because there are likely many compiler options that reference paths and some of those compiler options may be used outside of this ruleset (by those defining Scala libraries, binaries, tests, etc..), it's much less confusing to intercept compiler options than direct users to use another set of options instead.

I could go through every compiler option that references file paths, but that list is probably endless and I figured we can expand this list as necessary.